test/e2e/serial/tests/configuration.go

```diff
@@ -886,28 +933,59 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
 				Expect(err).ToNot(HaveOccurred())
 
 				if initialTopologyManagerPolicy != kcObj.TopologyManagerPolicy {
-					By("reverting kubeletconfig changes")
-					mcpsInfo, err := buildMCPsInfo(fxt.Client, ctx, nroOperObj)
-					Expect(err).ToNot(HaveOccurred())
-					Expect(mcpsInfo).ToNot(BeEmpty())
+					if numTargetedKCOwnerReferences == 0 {
+						By("reverting kuebeletconfig changes to the initial state under kubeletconfig")
+						mcpsInfo, err := buildMCPsInfo(fxt.Client, ctx, nroOperObj)
```

Comment:
let's be consistent and use same check flow in all the test, above in https://github.com/openshift-kni/numaresources-operator/pull/977/files#diff-99dd14d7f4fade18212bf91dbe6165d7626335bf407a28732eb7d591d2252fd8R900 the test checks the owner reference then it compares the initial vs current. here it compares the policies and then checks the owner reference let's decide on one way and use that. for example, we use the later in 47585 https://github.com/openshift-kni/numaresources-operator/pull/977/files#diff-99dd14d7f4fade18212bf91dbe6165d7626335bf407a28732eb7d591d2252fd8R900 and then the test calls waitForMcpUpdate once

Reply:
You're right it's mixed will choose one consistent way